pkg/controller/cluster/cluster_controller.go

```diff
@@ -660,6 +668,42 @@ func (c *clusterController) tryToFetchKubeSphereComponents(host string, transpor
 	return configz, nil
 }
 
+//
+func (c *clusterController) tryFetchKubeSphereVersion(host string, transport http.RoundTripper) (string, error) {
+	client := http.Client{
+		Transport: transport,
+		Timeout:   5 * time.Second,
+	}
+
+	response, err := client.Get(fmt.Sprintf(proxyFormat, host, "kapis/version"))
+	if err != nil {
+		return "", err
+	}
+
+	if response.StatusCode != http.StatusOK {
+		klog.V(4).Infof("Response status code isn't 200.")
+		return "", fmt.Errorf("response code %d", response.StatusCode)
+	}
+
+	info := version.Info{}
+	decoder := json.NewDecoder(response.Body)
+	err = decoder.Decode(&info)
+	if err != nil {
+		return "", err
+	}
+
+	//for kubesphere v3.0.0, the gitVersion is always v0.0.0, so we return v3.0.0
+	if info.GitVersion == "v0.0.0" {
```

Comment:
What about the previous version? Such 2.1?

Reply:
Currently, we don't support `Kubesphere v2.1` as a member cluster. @zryfish

Reply:
I'd suggest that return `unknown` instead of a possible value.

Reply:
`KubeSphere v2.1` will return 404 cause the version doesn't implement the `/kapis/version` API. Then we will not update the status.kubeSphereVersion filed. Do we still add an `unknown` value?

Reply:
Got it. How about adding these comments to the source file? It would be much more clear for other contributors.

Reply:
Writing comments is quite necessary. In my opinion, `KubeSphere v2.1` can not be joined as a member cluster, so it will never be reconciled. Maybe adding those comments will confuse other contributors.

Reply:
I added some comments. Hope it will be helpful.